homeassistant/components/history/__init__.py

```diff
@@ -46,18 +79,43 @@
 )
 
 
+@dataclass
+class HistoryLiveStream:
+    """Track a history live stream."""
+
+    stream_queue: asyncio.Queue[Event]
+    subscriptions: list[CALLBACK_TYPE]
+    end_time_unsub: CALLBACK_TYPE | None = None
+    task: asyncio.Task | None = None
+    wait_sync_task: asyncio.Task | None = None
+
+
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the history hooks."""
     conf = config.get(DOMAIN, {})
+    recorder_conf = config.get(RECORDER_DOMAIN, {})
+    history_conf = config.get(DOMAIN, {})
+    recorder_filter = extract_include_exclude_filter_conf(recorder_conf)
+    logbook_filter = extract_include_exclude_filter_conf(history_conf)
+    merged_filter = merge_include_exclude_filters(recorder_filter, logbook_filter)
+
+    possible_merged_entities_filter = convert_include_exclude_filter(merged_filter)
+
+    if not possible_merged_entities_filter.empty_filter:
+        hass.data[
+            HISTORY_FILTERS
+        ] = filters = sqlalchemy_filter_from_include_exclude_conf(conf)
+        hass.data[HISTORY_ENTITIES_FILTER] = possible_merged_entities_filter
+    else:
+        hass.data[HISTORY_FILTERS] = filters = None
+        hass.data[HISTORY_ENTITIES_FILTER] = None
 
-    hass.data[HISTORY_FILTERS] = filters = sqlalchemy_filter_from_include_exclude_conf(
-        conf
-    )
     hass.data[HISTORY_USE_INCLUDE_ORDER] = use_include_order = conf.get(CONF_ORDER)
```

Comment:
It would be nice to refactor this to use a dataclass, but I didn't want to do that in this PR

Reply:
I want to kill this feature so much.

Reply:
This feature was a thing back in the day when we loaded all entities at once on the history panel. And even then it was a bandaid.

Reply:
I think it can just go away since it's no longer a breaking change given there is no all history panel anymore.  I'll drop it when I convert this to a dataclass in a follow up

Reply:
https://github.com/home-assistant/core/pull/86365